tls/s2n_client_hello.c

```diff
@@ -452,6 +452,11 @@ int s2n_parse_client_hello(struct s2n_connection *conn)
     POSIX_GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
     POSIX_ENSURE_REF(ecc_pref);
     POSIX_ENSURE_GT(ecc_pref->count, 0);
+    /* When a curve is not negotiated between client and server, the server is free to choose any elliptic curves.
+     * Since we are setting the default curve to be p256, ecdhe ciphersuites can also be negotiated.
+     *
+     * https://www.rfc-editor.org/rfc/rfc4492#section-4
```

Comment:
The RFC section that you're adding a compliance comment for should be quoted (same with the test comment). Here's an example one: https://github.com/aws/s2n-tls/blob/7e1e08cbf3ad3e8fa30bce96365b2b7f6eeb1c73/tls/s2n_server_hello.c#L128-L133

Reply:
This block of code that sets the default curve already has comments associated with it, so I would consider just adding the duvet citation to the existing comment. And it might be good to make the RFC quote in the above comment duvet-compatible as well.

Also, you may need to modify some of the duvet stuff to get the new RFC to show up in the report. Duvet is run as a github action here: https://github.com/aws/s2n-tls/blob/main/.github/workflows/ci_compliance.yml, and the RFCs are added here: https://github.com/aws/s2n-tls/blob/main/compliance/initialize_duvet.sh.

Reply:
Also, you may want to either close this PR and open a new one for the comments or rename this one/modify the description.